src/test/java/newbank/TestApp.java

```diff
@@ -202,4 +200,61 @@ public void canExitApplication() throws IOException, InterruptedException {
     assertThat(response, containsString("SUCCESS"));
     client.join();
   }
+
+  @Test
+  public void canSetDefaultAccount() throws IOException {
+    setupCustomerWithAccount("TestCustomer4", "password4");
+
+    String response = testCommand("NEWACCOUNT Main DEFAULT\n");
+    assertThat(response, containsString("SUCCESS"));
+
+    response = testCommand("NEWACCOUNT Checking\n");
+    assertThat(response, containsString("SUCCESS"));
+
+    response = testCommand("DEFAULT Checking\n");
+    assertThat(response, containsString("SUCCESS"));
+
+    String result = testCommand("SHOWMYACCOUNTS\n");
+
+    String[] output = result.split(":");
+    assertThat(output[0].trim(), equalTo("Savings"));
+    assertThat(output[1].trim(), equalTo("0.00 GBP"));
+
+    result = display.getLine();
+    output = result.split(":");
+    assertThat(output[0].trim(), equalTo("Main"));
+    assertThat(output[1].trim(), equalTo("0.00 GBP"));
+
+    result = display.getLine();
+    output = result.split(":");
+    assertThat(output[0].trim(), equalTo("Checking*"));
+    assertThat(output[1].trim(), equalTo("0.00 GBP"));
+  }
+
+  @Test
+  public void handleSettingSavingsAsDefault() throws IOException {
+    setupCustomerWithAccount("TestCustomer5", "password5");
+
+    String response = testCommand("DEFAULT Savings\n");
+    assertThat(response, containsString("FAIL"));
```

Comment:
Perhaps the full error message here?

Reply:
I am only interested in the status of executing that command not the exact message which may vary.

Kindest Regards,
Jules

Reply:
Yes, I totally understand. Was just scared about false positives had another error be ever thrown.